### DISCUSSION TITLE
Tensorflow2 saved model workflow

### DISCUSSION THREAD

**Discussion by m-herrera** [nnstreamer/nnstreamer#3966] [Q&A]
Hi all,
I am trying to load a tensorflow v2 model with NNStreamer; however I get an error when loading the model. 
The error I get comes from this line: [L248](https://github.com/nnstreamer/nnstreamer/blob/ea2230e3855a47e1b75488af0b09ae3cc6aced02/ext/nnstreamer/tensor_filter/tensor_filter_tensorflow.cc#L248)

**Does NNstreamer tensorflow support loading the SavedModel format?** 
I believe the error happens because I'm using the SavedModel from tf v2 instead of the GraphDef from tf v1 directly.

I am building from source with the latest in the main branch.

Thanks in advance!

**Comment by myungjoo** [comment]
What's the error, then?
And does it work with its original model?
What about other (simple) saved models? (a model file to be tested?)

**Comment by m-herrera** [comment]
The error comes from [L248](https://github.com/nnstreamer/nnstreamer/blob/ea2230e3855a47e1b75488af0b09ae3cc6aced02/ext/nnstreamer/tensor_filter/tensor_filter_tensorflow.cc#L248), the message shown is from [L253](https://github.com/nnstreamer/nnstreamer/blob/ea2230e3855a47e1b75488af0b09ae3cc6aced02/ext/nnstreamer/tensor_filter/tensor_filter_tensorflow.cc#L253)

The specific error looks like the following:
```
** (gst-launch-1.0:11612): CRITICAL **: 07:32:35.162: Error deleting graph!! - [Code: 3] Invalid GraphDef

** (gst-launch-1.0:11612): CRITICAL **: 07:32:35.162: Failed to load model
Caught SIGSEGV
```

It can be reproduced using the example model found [here](https://www.tensorflow.org/guide/saved_model) and the following pipeline:
```
gst-launch-1.0 videotestsrc ! videoconvert ! video/x-raw,format=RGB,width=224,height=224 ! queue ! tensor_converter input-dim=1:224:224:3 ! tensor_filter input=1:224:224:3 inputtype=uint8 framework=tensorflow model=./saved_model.pb ! fakesink
```

I've been able to run the tensorflow examples without any issues, however these I believe don't use the SavedModel export format.

**Reply by myungjoo** [reply]
Could you please give me the direct link to the .pb file? not the instruction on how to make one?

**Reply by m-herrera** [reply]
I uploaded the saved model files [here](https://drive.google.com/drive/folders/1cB6xmDck5WQdITHneQjjKcfLcdPYpCfJ?usp=sharing). Tensorflow v2 exports a folder not only the `.pb` file, you can run the pipeline with the `.pb` file in that folder and should get the same error.
